lib/build/geo_feature_builder.dart

```diff
@@ -0,0 +1,155 @@
+import 'dart:io';
+import 'package:geojson/geojson.dart';
+import 'package:build/build.dart';
+import 'package:code_builder/code_builder.dart';
+import 'package:dart_style/dart_style.dart';
+import 'package:recase/recase.dart';
+
+class GeoFeatureBuilder implements Builder{
+
+  @override
+  Map<String, List<String>> get buildExtensions => {
+    options.config['source_file'] : ['lib/offline_geocoder/geo_feature_collection.dart']
+  };
+
+  final Map<String, String> propertiesToExtract = {};
+  late String dataFolderPath;
+  final List<Map<String, dynamic>> finalGeoFeatureList = [];
+  final BuilderOptions options;
+
+  GeoFeatureBuilder(this.options);
+
+  @override
+  Future<void> build(BuildStep buildStep) async {
+
+    final AssetId inputId = buildStep.inputId;
+
+    try {
+        final features = await featuresFromGeoJsonFile(File(inputId.path));
+        readConfiguration();
+        extractData(features);
+        final String code = createClass();
+        final lastSegment = inputId.uri.pathSegments.last;
+        final outputId = AssetId(inputId.package, inputId.path
+          .replaceFirst(inputId.path.toString().replaceAll(lastSegment, ''), 'lib/offline_geocoder/')
+          .replaceFirst(lastSegment, 'geo_feature_collection.dart'),
+        );
+        await buildStep.writeAsString(outputId, code);
+    } catch (e) {
+        print('Failed to generate class: $e');
+    }
+  }
+
+  void readConfiguration(){
+
+    final extractedProperties = options.config['extract_properties']
+                          .toString()
+                          .replaceAll('[', '')
+                          .replaceAll(']', '')
+                          .replaceAll('{', '')
+                          .replaceAll('}', '')
+                          .split(',');
+    for (final element in extractedProperties) {
+      final property = element.split(':');
+      if (property.length == 2){
+        propertiesToExtract.addAll({property[0].trim() : property[1].trim()});
+      }
+    }
+  }
```

Comment:
No implemented since after configuring the option as a Map can be assigned directly. However, I tried to use de json.decode but it does not work since I received the option value without the ""